### PR TITLE
Update to GHC 8.6.3

### PR DESCRIPTION
**Pull request by isovector**
This PR uses stack to pin the package dependencies to something reproducible, and updates `hbro` code so it works on modern `base`/other packages.

### CODE REVIEW

File: hbro.cabal [hunk 1]
```diff
@@ -22,53 +22,51 @@ Source-repository head
...
         template-haskell,
         text,
         -- text-icu,
-        time >= 1.5,
+        time ,
         transformers,
         transformers-base,
         unix,
         uuid,
-        webkitgtk3 >= 0.13.1.2,
```
Comment (k0ral) [inline hbro.cabal]: Why have you removed all lower-bounds ? They were protecting against known backward incompatibilities.

File: hbro.cabal [hunk 2]
```diff
@@ -22,53 +22,51 @@ Source-repository head
...
+        webkitgtk3,
         zeromq4-haskell
-    if impl(ghc < 7.10)
```
Comment (k0ral) [inline hbro.cabal]: Could you please:
- clarify what range of GHC versions is your change compatible with ? (is it only 8.6.3 ?)
- update the Travis CI configuration accordingly ?

File: library/Hbro/Boot.hs [hunk 3]
```diff
@@ -25,6 +32,8 @@ import           Hbro.Prelude
 import           Control.Concurrent.Async.Lifted
 import           Control.Concurrent.STM.MonadIO
 import           Control.Monad.Trans.Resource
+import           Control.Monad.Trans.Resource.Internal
```
Comment (k0ral) [inline library/Hbro/Boot.hs]: Is there no way to avoid importing this internal module from `resourcet` ?

File: library/Hbro/Boot.hs [hunk 4]
```diff
@@ -72,9 +99,11 @@ hbro settings = do
     case options of
         Left Rebuild -> Dyre.recompile >>= mapM_ (putStrLn . unpack)
         Left Version -> printVersions
-        Right runOptions -> runResourceT . runThreadedLoggingT (runOptions^.logLevel_) $ Dyre.wrap (runOptions^.dyreMode_)
```
Comment (k0ral) [inline library/Hbro/Boot.hs]: Unless I am mistaken, this is removing the dynamic configuration capability of `hbro`, which is a significant change of behavior. Is that intended ?

File: library/Hbro/Boot.hs (hunk 3 above)
Comment (isovector) [reply library/Hbro/Boot.hs]: Not without switching from `MonadBaseControl` to `MonadUnliftIO`, which is a bigger change than I want to make.

**Comment by isovector** [comment]
Thanks for the quick response on this! I'll admit the PR is pretty messy, I was just randomly fiddling around with it until I could get it to build. The only necessary changes should be the instances, though the `stack.yaml` is helpful. 

I'm not invested enough to put more work into this, but if you're keen, the instances need to be guarded via:

* `MonadBaseControl` + friends: `resourcet < 1.2.0`
* `Semigroup`: `base >= 4.11.0.0`

**Comment by k0ral** [comment]
I understand that you can't afford to invest more on this change, but I am not willing to merge it as is, so I will keep it open in case I have time to push it further in the future. But this may not be soon enough, to be honest, because I have stopped maintaining this project for a while.
Anyway, thank you for contributing.

**Pull request closed without merging**